TripKitAndroid/src/main/java/com/skedgo/tripkit/TripUpdaterImpl.kt

```diff
@@ -43,31 +44,36 @@ internal class TripUpdaterImpl(
                     response.body.processRawData(resources, gson)
                     emit(response.body.tripGroupList ?: listOf<TripGroup>())
                 }
+
                 else -> throw response.toException()
             }
         }.filter { it.isNotEmpty() }
-         .filter { it.firstOrNull()?.displayTrip != null }
-         .map { it.first().displayTrip!! }
+            .filter { it.firstOrNull()?.displayTrip != null }
+            .map { it.first().displayTrip!! }
```

Comment:
The map function is guaranteed not to produce any null values due to the preceding filter function. However, in line with Kotlin's safety conventions and to eliminate the need for using `!!`, we can consider an approach like:
```suggestion
            .filter { it.firstOrNull()?.displayTrip != null }
            .mapNotNull { it.firstOrNull()?.displayTrip }
```

Reply:
Hi @sg-jsonjuliane, I think we don't need to do it like this
`.filter { it.firstOrNull()?.displayTrip != null }`
since above it we're already doing `.filter { it.isNotEmpty() }` 
also as per checking we don't have `mapNotNull` in `io.reactivex.Observable` only `map`
Also only spacing was changed here due to reformatting of code. I suggest getting back to it later on.

Reply:
Hi @MichaelReyes, makes sense. Upon checking, the `mapNotNull` was an extension I created, maybe this is something I can introduce later on when I started working on the project. Thanks for the clarification.